view/latency.go

```diff
@@ -1,57 +1,62 @@
 	for i, result := range data.Results {
 		if i > 0 {
 			// new line as separator if more than 1 result
 			fmt.Println()
 		}
 
-		fmt.Fprintln(os.Stderr, generateHeader(result, ctx))
+		fmt.Fprintln(os.Stderr, generateProbeInfo(&result, !ctx.CI))
 
 		switch ctx.Cmd {
 		case "ping":
-			fmt.Println(latencyStatHeader("Min", ctx.CI) + fmt.Sprintf("%v ms", result.Result.Stats["min"]))
-			fmt.Println(latencyStatHeader("Max", ctx.CI) + fmt.Sprintf("%v ms", result.Result.Stats["max"]))
-			fmt.Println(latencyStatHeader("Avg", ctx.CI) + fmt.Sprintf("%v ms", result.Result.Stats["avg"]))
+			stats, err := client.DecodePingStats(result.Result.StatsRaw)
+			if err != nil {
+				return err
+			}
+			fmt.Println(latencyStatHeader("Min", ctx.CI) + fmt.Sprintf("%.2f ms", stats.Min))
+			fmt.Println(latencyStatHeader("Max", ctx.CI) + fmt.Sprintf("%.2f ms", stats.Max))
+			fmt.Println(latencyStatHeader("Avg", ctx.CI) + fmt.Sprintf("%.2f ms", stats.Avg))
 		case "dns":
-			timings, err := client.DecodeTimings(ctx.Cmd, result.Result.TimingsRaw)
+			timings, err := client.DecodeDNSTimings(result.Result.TimingsRaw)
 			if err != nil {
-				fmt.Println(err)
-				return
+				return err
 			}
-			fmt.Println(latencyStatHeader("Total", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["total"]))
+			fmt.Println(latencyStatHeader("Total", ctx.CI) + fmt.Sprintf("%v ms", timings.Total))
 		case "http":
-			timings, err := client.DecodeTimings(ctx.Cmd, result.Result.TimingsRaw)
+			timings, err := client.DecodeHTTPTimings(result.Result.TimingsRaw)
 			if err != nil {
-				fmt.Println(err)
-				return
+				return err
 			}
-			fmt.Println(latencyStatHeader("Total", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["total"]))
-			fmt.Println(latencyStatHeader("Download", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["download"]))
-			fmt.Println(latencyStatHeader("First byte", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["firstByte"]))
-			fmt.Println(latencyStatHeader("DNS", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["dns"]))
-			fmt.Println(latencyStatHeader("TLS", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["tls"]))
-			fmt.Println(latencyStatHeader("TCP", ctx.CI) + fmt.Sprintf("%v ms", timings.Interface["tcp"]))
+			fmt.Println(latencyStatHeader("Total", ctx.CI) + fmt.Sprintf("%v ms", timings.Total))
+			fmt.Println(latencyStatHeader("Download", ctx.CI) + fmt.Sprintf("%v ms", timings.Download))
+			fmt.Println(latencyStatHeader("First byte", ctx.CI) + fmt.Sprintf("%v ms", timings.FirstByte))
+			fmt.Println(latencyStatHeader("DNS", ctx.CI) + fmt.Sprintf("%v ms", timings.DNS))
+			fmt.Println(latencyStatHeader("TLS", ctx.CI) + fmt.Sprintf("%v ms", timings.TLS))
+			fmt.Println(latencyStatHeader("TCP", ctx.CI) + fmt.Sprintf("%v ms", timings.TCP))
 		default:
-			panic("unexpected command for latency output: " + ctx.Cmd)
+			return errors.New("unexpected command for latency output: " + ctx.Cmd)
 		}
-
 	}
 
 	if ctx.Share {
```

Comment:
For `--share` combined with `--infinite`, the behavior should be:

 - When the process ends, print a link in the form `https://www.jsdelivr.com/globalping?measurement=id1+id2+id3` where all the IDs are ordered from oldest to newest
 - If there are more than 10 measurements, include only the 10 most recent IDs in the link, and show a warning below the link: "For long-running continuous mode measurements, only the last 160 packets are shared."

Reply:
The site doesn't seem to work with multiple ids.

Reply:
Not yet, but we're working on adding support for this format.